README.md

```diff
@@ -3,3 +3,177 @@
 Simple declarative data layer for React applications
 
 [![Build Status](https://travis-ci.org/dprokop/querier.svg?branch=master)](https://travis-ci.org/dprokop/querier)
+
+# Table of Content
+  * [What is Querier?](#what-is-querier)
+  * [What is a declarative data layer?](#what-is-declarative-data-layer)
+  * [Querier GIST](#querier-gist)
+  * [Queries](#queries)
+    * [Input queries](#input-queries)
+    * [Action queries](#action-queries)
+  * [Handling query statuses](#handling-query-statuses)
+  * [Caching](#caching)
+  * [Server-side rendering](#server-side-rendering)
+  * [API](#server-side-rendering)
+  * [Known limitations](#known-limitations)
+
+## What is Querier?
+Querier is a tool that helps you get remote and async data into React app effortlessly.
+
+With simple, declarative API, you don't have to worry about handling async state or errors. All you have to do is decorate component that requires async data with `withData` higher-order component. Then, results, states and possible errors will be injected into the component as props.
+
+Querier is implemented and tested in TypeScript.
+
+## What is a declarative data layer?
+Simply, it is a way to supply data to your components. You declare component's data requirements, and the whole logic for fetching it, supplying status and results is handled by Querier itself. To achieve that, you implement `queries`. Queries are simply **async functions**. They can either communicate with remote source or resolve async data in whatever way you imagine.
+
+## Querier Gist
+```tsx
+import * as React from 'react';
+import { getRepository, starRepository, RepositoryType } from 'app/data/octokit';
+import {  withData,  WithDataProps,  QuerierState } from 'querier';
+
+type RepositoryDetailsOwnProps = {  id: number }
+type RepositorDetailsInputQueries = {  repository: RepositoryType }
+type RepositorDetailsActionQueries = {  starRepository: RepositoryType }
+type RepositoryDetailsProps = WithDataProps<
+  RepositoryDetailsOwnProps,
+  RepositoryDetailsInputQueries,
+  RepositorDetailsActionQueries
+>;
```

Comment:
Maybe some comment why are those types needed for

Reply:
Ready in 7f2c1eee5b7081eb6cac1dfc6071ed439efa8c69